pkg/bootstrap/bootstrap_config.go

```diff
@@ -268,6 +268,11 @@ func WriteBootstrap(config *meshconfig.ProxyConfig, node string, epoch int, pilo
 	// Support multiple network interfaces
 	meta["ISTIO_META_INSTANCE_IPS"] = strings.Join(nodeIPs, ",")
 
+	if config.Sds != nil && config.Sds.Enabled {
```

Comment:
I think you can pass the opts directly from caller (based on detected value), without using config.Sds ?